Add docs on DR (#5797)
dimagi/commcare-cloud: hosting_docs/source/backups_dr/4-dr-setup.rst

> Disaster Recovery Setup

Reviewer: Just something small I noticed, not needed to rectify if you don't want to: You removed the "Setup" part from "Disaster Recovery Setup", but it's still in the name of the file :smile:

Author: Oh, yeah. I guess this is okay. I removed from the title since the doc talks a bit more than DR setup.

=================

.. contents:: Table of Contents
    :depth: 2

Overview
--------

> Disaster Recovery refers to the process of restoring the IT services following an event of outage of IT infrastrcture due to a natural or human induced disaster to allow business continuity. This `public artcile <https://www.ibm.com/in-en/topics/backup-disaster-recovery>`_ from IBM is useful to understand what Disaster Recovery is in general.

Reviewer 2: Typo
```suggestion
Disaster Recovery refers to the process of restoring the IT services following an event of outage of IT infrastructure due to a natural or human induced disaster to allow business continuity. This `public article <https://www.ibm.com/in-en/topics/backup-disaster-recovery>`_ from IBM is useful to understand what Disaster Recovery is in general.
```

A Disaster Recovery solution at a minimum involves

*
  Establishing Recovery Time and Recovery Point Objectives that meet project requirements.

*
  Setting up and monitoring remote data backups or replication.

*
  Creating an active or passive secondary site and the necessary automation to failover and restore to this site rapidly.

> Since Commcare mobile works offline, a disaster at primary infrastructure may not cause an immediate disruption to mobile worker operations. But it definitely impacts every web operations (for e.g. actions based on reports, SMS reminders and integrations) and will soon clog mobile worker operations as well. 

Reviewer 2: Typo

```suggestion
Since CommCare Mobile works offline, a disaster at primary infrastructure may not cause an immediate disruption to mobile worker operations. But it definitely impacts all web operations (for e.g. actions based on reports, SMS reminders and integrations) and will soon clog mobile worker operations as well. 
```

> Since Commcare mobile works offline, a disaster at primary infrastructure may not cause an immediate disruption to mobile worker operations. But it definitely impacts every web operations (for e.g. actions based on reports, SMS reminders and integrations) and will soon clog mobile worker operations as well. 

Reviewer 3: ```suggestion
Since CommCare mobile works offline, a disaster at primary infrastructure may not cause an immediate disruption to mobile worker operations. But it definitely impacts all web operations (for e.g. actions based on reports, SMS reminders and integrations) and will soon clog mobile worker operations as well. 
```

If not, you can create another environment directory with updated configuration based on the primary config directory.

Remote Backups
--------------

The commcare-cloud backup tool provides everything to enable remote backups. It takes daily and weekly backups by default and can be configured to take hourly backups. The tool can send backups to a remote S3 compatible service if configured. The documenation on :ref:`backups` has the relevant details on how to setup remote backups and restore from backups.

While the remote backups are the minimum requirement to safeguard the data in the event of a disaster it is not enough for a rapid recovery of the CommCare services. Even though, commcare-cloud is able to provide hourly backups it will not be enough to achive one hour RPO since it is not possible to setup a CommCare environment rapidly in an hour at a secondary site. To enable rapid recovery you must have a secondary environment on standby, setup database replication and a failproof recovery plan and scripts.

Database Replication
--------------------

> Continous Database replication is necessary to enable minimal RPO. While commcare-cloud does not have tooling available to setup continous replication all the databases used in CommCareHQ support replication through various means. To set this up, you should consult database specific documentation and set this up yourself.

Reviewer: What is `RPO`?

Reviewer 2: ```suggestion
Continous Database replication is necessary to enable minimal RPO. While commcare-cloud does not have tooling available to set up continous replication all the databases used in CommCareHQ support replication through various means. To set this up, you should consult database specific documentation and set this up yourself.
```

Author: RPO is Recovery Point Objective. This is not abbreviated here with the assumption that the reader has consulted the background docs on DR that were mentioned above.

Primary Databases

*
  Postgresql https://www.postgresql.org/docs/current/runtime-config-replication.html

*
  BlobDB If you are using MinIO https://min.io/product/active-data-replication-for-object-storage

*
  CouchDB https://docs.couchdb.org/en/stable/replication/intro.html

Secondary Databases

* Elasticsearch https://www.elastic.co/guide/en/cloud-enterprise/2.4/ece-snapshots.html#ece-restore-across-clusters

Example models
--------------

Below are some example models for DR.

*

>   Remote backups on secondary environment: You can setup a CommcareHQ environment on secondary infrastructure, keep it upto date with remote backups and keep restore scripts ready.

Reviewer 2: ```suggestion
  Remote backups on secondary environment: You can set up a CommCare HQ environment on secondary infrastructure, keep it up to date with remote backups and keep restore scripts ready.
```

*

>   Secondary environment with contious data replication: You can setup a fully functioning secondary environment with databases being replicated continously.

Reviewer: Small spelling issue: `contious`

Author: Updated
